src/FrameworkBridge/Symfony/CacheWarmer/ProxyCacheWarmer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenClassrooms\ServiceProxy\FrameworkBridge\Symfony\CacheWarmer;
+
+use ProxyManager\Proxy\LazyLoadingInterface;
+use Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerInterface;
+use Symfony\Component\VarExporter\LazyObjectInterface;
+
+final class ProxyCacheWarmer implements CacheWarmerInterface
+{
+    /**
+     * @var iterable<object>
+     */
+    private iterable $proxies;
+
+    /**
+     * @param iterable<object> $proxies
+     */
+    public function __construct(iterable $proxies)
+    {
+        $this->proxies = $proxies;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp(string $cacheDir): array
+    {
+        foreach ($this->proxies as $proxy) {
+            if ($proxy instanceof LazyLoadingInterface && !$proxy->isProxyInitialized()) {
+                $proxy->initializeProxy();
+            }
+
+            if (class_exists(LazyObjectInterface::class)
+                && $proxy instanceof LazyObjectInterface
+                && !$proxy->isLazyObjectInitialized()) {
+                $proxy->initializeLazyObject();
+            }
+        }
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isOptional()
+    {
+        return true;
```

Comment:
maybe false ?

Reply:
The isOptional() method should return true if it's possible to use the application without calling this cache warmer. In Symfony, optional warmers are always executed by default (you can change this by using the --no-optional-warmers option when executing the command).